### DISCUSSION TITLE
If cookie exist function

### DISCUSSION THREAD

**Discussion by MaxAllianov** [cypress-io/cypress#18376] [Questions and Help] [answered]
How can I make the If function which checks if the cookie exists using the cypress?
I have tried this, but unfortunately, the test is failing on the should assertion.
I'm very new in the cypress, any advice would be helpful

```
 before('Login', () => {
        cy.visit('/')
        cy.getCookies().then((cookies) => {
          if 
          (cy.getCookie('auth').should('exist')) 
            {
            cy.log('Already authorisated')
            }
          else 
            {
              //something
            }
        })    
    }) 
```

![image](https://user-images.githubusercontent.com/76516974/136216911-0fa99a04-697e-44c2-a2c4-afa914eeb76c.png)

**Comment by bahmutov** [comment, marked as answer]
Take a look at https://glebbahmutov.com/cypress-examples/8.5.0/recipes/conditional-testing.html#use-a-cookie-if-present - but I think you want to use something like https://github.com/bahmutov/cypress-data-session
